src/website/webview/views.py

```diff
@@ -547,6 +612,8 @@ def get_context_data(self, **kwargs: Any) -> Any:
 
         for obj in context["object_list"]:
             obj.packages = channel_structure(obj.derivations.all())
+            obj.activity_log = get_suggestion_activity_log(obj)
```

Comment:
this is inefficient, you should have `get_suggestion_activity_logs(<list of object>)` and returns a map of `{pk: logs}` and then interleave them there.

Reply:
From https://github.com/Nix-Security-WG/nix-security-tracker/pull/366#issuecomment-2487949573
>Eventually we probably want to not load the logs for the entire list, but only on demand 

@fricklerhandwerk Is this the case? Because in that case I'd skip this optimization, since it's the current logic is already tailored to "gather the activity log of this suggestion".